pele_platform/analysis/analysis.py

```diff
@@ -1541,4 +1698,128 @@ def _check_residue_exists(self):
             raise ValueError(
                 f"Residue {self.residue} was not found in output "
                 f"trajectories. Make sure you are passing a correct "
-                f"'resname' argument to Analysis.")
+                f"'resname' argument to Analysis."
+            )
+
+    def _run_inner_clustering(
+        self,
+        path,
+        selected_clusters,
+        all_clusters,
+        coordinates,
+        representatives_criterion,
+        dataframe,
+    ):
+        """
+        Performs k-means clustering to provide the user with representatives of each selected cluster.
+
+        Parameters
+        ----------
+        selected_clusters : a numpy.array object
+            The array of cluster labels that were obtained
+        path : str
+            The path where the clusters will be saved.
+        representatives_criterion : str
+            Criterion "multi n" where n is the number of expected clusters.
+        coordinates : np.array
+            Array with all coordinates used for initial clustering. These need to be filtered to only include the ones
+            belonging to selected clusters.
+        dataframe : a pandas.dataframe object
+            The dataframe containing the PELE reports information that
+            follows the same ordering as the array of clusters
+        """
+        from collections import defaultdict
+        import numpy as np
+        import os
+        import pandas as pd
+        from sklearn.cluster import KMeans
+
+        from pele_platform.Errors.custom_errors import ClusteringError
+        from pele_platform.analysis.clustering import get_cluster_label
+        from pele_platform.Utilities.Helpers.bestStructs import (
+            extract_snapshot_from_pdb,
+            extract_snapshot_from_xtc,
+        )
+        from pele_platform.Utilities.Helpers import get_suffix
+
+        random_seed = 42 if not self.skip_initial_structures else None  # needed for reproducible tests
```

Comment:
Why dont we take the same seed as for PELE (the one initialized in the input yaml)? It should be another parameter for analysis, then, but I think it is worth.